src/plugins/intel_gna/src/transformations/gather_sinking_transpose_reshape.cpp

```diff
@@ -19,154 +19,120 @@
+    const auto gather_indices_value =
+        graph_utils::make_gather_indexes_from_transpose_axes(transpose->get_input_shape(0),
+                                                             transpose_const->get_axis_vector_val());
+    ov::Shape shape_out = reshape->get_output_shape(0);
+
+    // Set Gather input shape
+    std::vector<int8_t> reshape_in_dims = {1, -1};
+    for (size_t i = reshape_in_dims.size(); i < shape_out.size(); ++i) {
+        if (shape_out[i] == 1) {
+            reshape_in_dims.push_back(1);
         }
     }
+    // reshape
+    auto reshape_in_const =
+        std::make_shared<Constant>(ov::element::i64, ov::Shape{reshape_in_dims.size()}, reshape_in_dims);
+    auto reshape_in = reshape->clone_with_new_inputs({transpose->input_value(0), reshape_in_const});
+    new_nodes.push_back(reshape_in);
 
-    return gather_order;
-}
-
-NodePair SinkForward(NodePtr transpose, std::shared_ptr<Constant> transpose_constant, NodePtr reshape) {
-    const auto gather_indices_value =
-        CreateGatherIndices(transpose->get_input_shape(0), transpose_constant->get_axis_vector_val());
-
-    const int64_t gather_axis_value = graph_utils::get_first_valuable_dim_id(reshape->get_output_shape(0));
-
-    auto reshape_new = reshape->clone_with_new_inputs({transpose->input_value(0), reshape->input_value(1)});
-
+    const int64_t gather_axis_value = graph_utils::get_first_valuable_dim_id(reshape_in->get_output_shape(0));
     auto gather_axis = std::make_shared<Constant>(ov::element::i64, ov::Shape{}, gather_axis_value);
     auto gather_indices =
         std::make_shared<Constant>(ov::element::i64, ov::Shape{gather_indices_value.size()}, gather_indices_value);
-    auto gather = std::make_shared<Gather>(reshape_new, gather_indices, gather_axis);
+    auto gather = std::make_shared<Gather>(reshape_in, gather_indices, gather_axis);
+    new_nodes.push_back(gather);
 
-    ov::replace_node(reshape, gather);
+    auto reshape_out_const = std::make_shared<Constant>(ov::element::i64, ov::Shape{shape_out.size()}, shape_out);
+    auto reshape_out = std::make_shared<Reshape>(gather, reshape_out_const, false);
 
-    ov::copy_runtime_info({reshape}, {gather, gather_indices, gather_axis, reshape_new});
-    gather->set_friendly_name(reshape->get_friendly_name());
+    if (!graph_utils::are_shapes_equal(reshape_out->get_input_shape(0), reshape_out->get_output_shape(0))) {
+        new_nodes.push_back(reshape_out);
+    }
 
-    return std::make_pair(reshape_new, gather);
+    ov::replace_output_update_name(reshape->output(0), new_nodes.back()->output(0));
+    ov::copy_runtime_info({reshape}, new_nodes);
+
+    return new_nodes;
 }
 
-NodePair SinkBackward(NodePtr transpose, std::shared_ptr<Constant> transpose_constant, NodePtr reshape) {
-    const int64_t gather_axis_value = graph_utils::get_first_valuable_dim_id(reshape->get_input_shape(0));
+std::vector<std::shared_ptr<ov::Node>> SinkBackward(std::shared_ptr<ov::Node> reshape,
```

Comment:
Please use function naming convention here and in other places - lower case with underscore.